### ISSUE TITLE
Series.value_counts throws AssertionError with type int8 

### ISSUE THREAD

**Issue by randerzander** [gpuopenanalytics/pygdf#144]
```
import pandas as pd

df = pd.DataFrame(data = {'num_col': [1, 2], 'cat_col': ['a', 'b']})
df.cat_col = df.cat_col.astype('category')
df['cat_codes'] = df.cat_col.cat.codes
# Work around -- cast the codes column to a type that works
#df['cat_codes'] = df.cat_col.cat.codes.astype('int64')

from pygdf.dataframe import DataFrame as gdf
my_gdf = gdf.from_pandas(df)
my_gdf.cat_codes.value_counts()
```
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-60-2feb35565298> in <module>()
----> 1 my_gdf.cat_codes.value_counts()

/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a2+168.g78a5ca8-py3.6.egg/pygdf/series.py in value_counts(self, type)
    735         if self.null_count == len(self):
    736             return 0
--> 737         vals, cnts = self._column.value_counts(type=type)
    738         res = Series(cnts, index=GenericIndex(vals))
    739         return res

/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a2+168.g78a5ca8-py3.6.egg/pygdf/numerical.py in value_counts(self, type)
    167         out2 = cudautils.value_count(segs, len(sortedvals))
    168         out_vals = self.replace(data=Buffer(out1), mask=None)
--> 169         out_counts = self.replace(data=Buffer(out2), mask=None)
    170         return out_vals, out_counts
    171 

/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a2+168.g78a5ca8-py3.6.egg/pygdf/column.py in replace(self, **kwargs)
    282         if 'mask' in kwargs and 'null_count' not in kwargs:
    283             del params['null_count']
--> 284         return type(self)(**params)
    285 
    286     def view(self, newcls, **kwargs):

/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a2+168.g78a5ca8-py3.6.egg/pygdf/numerical.py in __init__(self, **kwargs)
     60         """
     61         super(NumericalColumn, self).__init__(**kwargs)
---> 62         assert self._dtype == self._data.dtype
     63 
     64     def serialize(self, serialize):

AssertionError: 
```

**Comment by dantegd** [comment]
Yes, this was solved with #153 merge, just as issue 145

**Comment by kkraus14** [comment]
Great, closing.
